### PR TITLE
cudatoolkit: reintroduce version 11.7.0 to master

### PR DESCRIPTION
**Pull request by dguibert**
###### Description of changes

This patch fixes https://github.com/NixOS/nixpkgs/pull/179912 which has been reverted on master due to https://gist.github.com/GrahamcOfBorg/45ac7f5bc9e02a74cb1e4264f365417f

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix [hunk 1]
```diff
@@ -37,8 +37,12 @@ in (lib.filterAttrs (attr: _: (prev ? "${attr}")) {
...
 
   nsight_compute = prev.nsight_compute.overrideAttrs (oldAttrs: {
-    nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ pkgs.qt5.wrapQtAppsHook ];
-    buildInputs = oldAttrs.buildInputs ++ [ pkgs.libsForQt5.qt5.qtwebview ];
+    nativeBuildInputs = oldAttrs.nativeBuildInputs
+      ++ lib.optionals (lib.versionOlder   prev.nsight_compute.version "2022.2.0") [ pkgs.qt5.wrapQtAppsHook ]
+      ++ lib.optionals (lib.versionAtLeast prev.nsight_compute.version "2022.2.0") [ pkgs.qt6.wrapQtAppsHook ];
+      buildInputs = oldAttrs.buildInputs
+      ++ lib.optionals (lib.versionOlder   prev.nsight_compute.version "2022.2.0") [ pkgs.qt5.qtwebview ]
+      ++ lib.optionals (lib.versionAtLeast prev.nsight_compute.version "2022.2.0") [ pkgs.qt6.qtwebview ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/cudatoolkit/redist/overrides.nix:45]: I think an if else would be actually nicer here.

File: pkgs/test/cuda/cuda-samples/extension.nix [hunk 2]
```diff
@@ -11,6 +11,9 @@ final: prev: let
     "11.4" = "082dkk5y34wyvjgj2p5j1d00rk8xaxb9z0mhvz16bd469r1bw2qk";
     "11.5" = "sha256-AKRZbke0K59lakhTi8dX2cR2aBuWPZkiQxyKaZTvHrI=";
     "11.6" = "sha256-AsLNmAplfuQbXg9zt09tXAuFJ524EtTYsQuUlV1tPkE=";
+    # the tag 11.7 does not exists: see https://github.com/NVIDIA/cuda-samples/issues/128
+    # maybe fixed by https://github.com/NVIDIA/cuda-samples/pull/133
+    "11.7" = prev.lib.fakeSha256;
```
Comment (SuperSandro2000) [inline pkgs/test/cuda/cuda-samples/extension.nix:16]: but the fakeSha will not work.

Comment (dguibert) [reply pkgs/test/cuda/cuda-samples/extension.nix:16]: What do you suggest to handle the nonexistence of the tag?

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix (hunk 1 above)
Comment (dguibert) [reply pkgs/development/compilers/cudatoolkit/redist/overrides.nix:45]: Yes, maybe on another PR or do you prefer in this one?

File: pkgs/development/libraries/science/math/cudnn/extension.nix [hunk 3]
```diff
@@ -94,7 +94,23 @@ final: prev: let
         fullVersion = "8.3.2.44";
         hash = "sha256-VQCVPAjF5dHd3P2iNPnvvdzb5DpTsm3AqCxyP6FwxFc=";
         url = "${urlPrefix}/v${majorMinorPatch fullVersion}/local_installers/${fileVersion}/cudnn-linux-x86_64-${fullVersion}_cuda${fileVersion}-archive.tar.xz";
-        supportedCudaVersions = [ "11.0" "11.1" "11.2" "11.3" "11.4" "11.4" "11.5" "11.6" ];
+        supportedCudaVersions = [ "11.0" "11.1" "11.2" "11.3" "11.4" "11.4" "11.5" "11.6" "11.7" ];
```
Comment (samuela) [inline pkgs/development/libraries/science/math/cudnn/extension.nix:97]: duplicate 11.4

File: pkgs/development/libraries/science/math/cudnn/extension.nix [hunk 4]
```diff
@@ -94,7 +94,23 @@ final: prev: let
...
+        hash = "sha256-FMXjykJYJxmW0f2VnELRfFgs5Nmv9FH4RSRGnnhP0VQ=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/local_installers/${fileVersion}/cudnn-linux-x86_64-${fullVersion}_cuda${fileVersion}-archive.tar.xz";
+        supportedCudaVersions = [ "10.2" ];
+      }
+      rec {
+        fileVersion = "11.6";
+        fullVersion = "8.4.0.27";
+        hash = "sha256-0Zva/ZgAx50p5vb/+p+eLBDREy1sL/ELFZPgV+dN0FA=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/local_installers/${fileVersion}/cudnn-linux-x86_64-${fullVersion}_cuda${fileVersion}-archive.tar.xz";
+        supportedCudaVersions = [ "11.0" "11.1" "11.2" "11.3" "11.4" "11.4" "11.5" "11.6" "11.7" ];
```
Comment (samuela) [inline pkgs/development/libraries/science/math/cudnn/extension.nix:113]: duplicate 11.4

**Review by samuela — CHANGES_REQUESTED** [review]
Thanks for resubmitting this @dguibert and apologies that this has been more involved than expected!

I think these changes look good, but I'd actually like us to go a bit further and de-couple TensorRT and cudatoolkit evaluation! It's totally fine for an upgrade to one of the packages to end up marking other derivations as broken, but breaking evaluation is a huge problem. Having to constantly keep the two in sync is an unnecessary burden on both packages and really came back to bite us in https://github.com/NixOS/nixpkgs/pull/179912. 

Could we refactor `pkgs/development/libraries/science/math/tensorrt/extension.nix` a bit such that upgrading cudatoolkit can be done without breaking the evaluation of tensorrt?

**Comment by aidalgol** [comment]
> Could we refactor `pkgs/development/libraries/science/math/tensorrt/extension.nix` a bit such that upgrading cudatoolkit can be done without breaking the evaluation of tensorrt?

@samuela Sorry, I only just saw this PR.  Since I introduced TensorRT to nixpkgs, I feel I should take this on.  Could you elaborate a bit on what exactly you're proposing?

**Comment by samuela** [comment]
Hi @aidalgol, I think you nailed it in https://github.com/NixOS/nixpkgs/pull/192958. IIRC the issue was that TensorRT's evaluation would break if `cudaVersion` was upgraded without editing TensorRT's derivation. But it looks like https://github.com/NixOS/nixpkgs/pull/192958 should fix that!

**Comment by samuela** [comment]
@dguibert If you rebase off of https://github.com/NixOS/nixpkgs/pull/192958, then I think this should be good to go. Note that adding 11.7 to the supported versions of cudnn and TensorRT is still necessary in order to prevent those packages from being marked `meta.broken` but we shouldn't have any evaluation issues from here on out (thanks @aidalgol!!)

File: pkgs/test/cuda/cuda-samples/extension.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/test/cuda/cuda-samples/extension.nix:16]: Use a commit hash or replace it with a throw.

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/compilers/cudatoolkit/redist/overrides.nix:45]: Why not to it in this PR? Reduces the amount of rebuilds people need to do.

File: pkgs/test/cuda/cuda-samples/extension.nix (hunk 2 above)
Comment (dguibert) [reply pkgs/test/cuda/cuda-samples/extension.nix]: done

File: pkgs/development/libraries/science/math/cudnn/extension.nix (hunk 3 above)
Comment (dguibert) [reply pkgs/development/libraries/science/math/cudnn/extension.nix]: remove duplicate

File: pkgs/development/libraries/science/math/cudnn/extension.nix (hunk 4 above)
Comment (dguibert) [reply pkgs/development/libraries/science/math/cudnn/extension.nix]: removed duplicate.

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix (hunk 1 above)
Comment (dguibert) [reply pkgs/development/compilers/cudatoolkit/redist/overrides.nix]: done in the last commit

**Comment by dguibert** [comment]
Hi @samuela and @SuperSandro2000,
I took time to rebase this PR and apply your suggestions ;-)

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix [hunk 5]
```diff
@@ -37,8 +37,14 @@ in (lib.filterAttrs (attr: _: (prev ? "${attr}")) {
   ];
 
   nsight_compute = prev.nsight_compute.overrideAttrs (oldAttrs: {
-    nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ pkgs.qt5.wrapQtAppsHook ];
-    buildInputs = oldAttrs.buildInputs ++ [ pkgs.libsForQt5.qt5.qtwebview ];
+    nativeBuildInputs = oldAttrs.nativeBuildInputs
+    ++ (if (lib.versionOlder   prev.nsight_compute.version "2022.2.0")
```
Comment (samuela) [inline pkgs/development/compilers/cudatoolkit/redist/overrides.nix:41]:
```suggestion
    ++ (if (lib.versionOlder prev.nsight_compute.version "2022.2.0")
```

File: pkgs/development/compilers/cudatoolkit/redist/overrides.nix [hunk 6]
```diff
@@ -37,8 +37,14 @@ in (lib.filterAttrs (attr: _: (prev ? "${attr}")) {
...
+       then [ pkgs.qt5.wrapQtAppsHook ]
+       else [ pkgs.qt6.wrapQtAppsHook ]);
+    buildInputs = oldAttrs.buildInputs
+    ++ (if (lib.versionOlder   prev.nsight_compute.version "2022.2.0")
```
Comment (samuela) [inline pkgs/development/compilers/cudatoolkit/redist/overrides.nix:45]:
```suggestion
    ++ (if (lib.versionOlder prev.nsight_compute.version "2022.2.0")
```

**Review by samuela — COMMENTED** [review]
Nice! I think this looks good overall. Running nixpkgs-review now...

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 185557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>cudaPackages.nvidia_driver</li>
    <li>python310Packages.cupy</li>
    <li>python39Packages.cupy</li>
    <li>truecrack-cuda</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>cudaPackages.tensorrt (cudaPackages.tensorrt_8_4_0)</li>
    <li>ethminer (ethminer-cuda)</li>
    <li>mathematica-cuda</li>
    <li>python310Packages.numbaWithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.tensorrt</li>
    <li>python39Packages.numbaWithCuda</li>
    <li>python39Packages.pyrealsense2WithCuda</li>
    <li>python39Packages.tensorrt</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>colmapWithCuda</li>
    <li>cudaPackages.cuda_cccl</li>
    <li>cudaPackages.cuda_cudart</li>
    <li>cudaPackages.cuda_cuobjdump</li>
    <li>cudaPackages.cuda_cupti</li>
    <li>cudaPackages.cuda_cuxxfilt</li>
    <li>cudaPackages.cuda_demo_suite</li>
    <li>cudaPackages.cuda_documentation</li>
    <li>cudaPackages.cuda_gdb</li>
    <li>cudaPackages.cuda_memcheck</li>
    <li>cudaPackages.cuda_nsight</li>
    <li>cudaPackages.cuda_nvcc</li>
    <li>cudaPackages.cuda_nvdisasm</li>
    <li>cudaPackages.cuda_nvml_dev</li>
    <li>cudaPackages.cuda_nvprof</li>
    <li>cudaPackages.cuda_nvprune</li>
    <li>cudaPackages.cuda_nvrtc</li>
    <li>cudaPackages.cuda_nvtx</li>
    <li>cudaPackages.cuda_nvvp</li>
    <li>cudaPackages.cuda_sanitizer_api</li>
    <li>cudatoolkit (cudaPackages.cudatoolkit ,cudatoolkit_11)</li>
    <li>cudaPackages.cudnn (cudaPackages.cudnn_8_4_0)</li>
    <li>cudaPackages.cudnn_8_3_2</li>
    <li>cudaPackages.cutensor</li>
    <li>cudaPackages.fabricmanager</li>
    <li>cudaPackages.libcublas</li>
    <li>cudaPackages.libcufft</li>
    <li>cudaPackages.libcufile</li>
    <li>cudaPackages.libcurand</li>
    <li>cudaPackages.libcusolver</li>
    <li>cudaPackages.libcusparse</li>
    <li>cudaPackages.libnpp</li>
    <li>cudaPackages.libnvidia_nscq</li>
    <li>cudaPackages.libnvjpeg</li>
    <li>cudaPackages.nccl</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cudaPackages.nsight_systems</li>
    <li>cudaPackages.nvidia_fs</li>
    <li>forge</li>
    <li>gpu-burn</li>
    <li>gromacsCudaMpi</li>
    <li>gwe</li>
    <li>katagoWithCuda</li>
    <li>librealsenseWithCuda</li>
    <li>magma</li>
    <li>nvtop</li>
    <li>nvtop-nvidia</li>
    <li>python310Packages.TheanoWithCuda</li>
    <li>python310Packages.pycuda</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python39Packages.TheanoWithCuda</li>
    <li>python39Packages.pycuda</li>
    <li>python39Packages.pynvml</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.torchWithCuda</li>
    <li>xgboostWithCuda</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 185557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>cudaPackages.nvidia_driver</li>
    <li>python310Packages.cupy</li>
    <li>python39Packages.cupy</li>
    <li>truecrack-cuda</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>cudaPackages.tensorrt (cudaPackages.tensorrt_8_4_0)</li>
    <li>ethminer (ethminer-cuda)</li>
    <li>mathematica-cuda</li>
    <li>python310Packages.numbaWithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.tensorrt</li>
    <li>python39Packages.numbaWithCuda</li>
    <li>python39Packages.pyrealsense2WithCuda</li>
    <li>python39Packages.tensorrt</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>colmapWithCuda</li>
    <li>cudaPackages.cuda_cccl</li>
    <li>cudaPackages.cuda_cudart</li>
    <li>cudaPackages.cuda_cuobjdump</li>
    <li>cudaPackages.cuda_cupti</li>
    <li>cudaPackages.cuda_cuxxfilt</li>
    <li>cudaPackages.cuda_demo_suite</li>
    <li>cudaPackages.cuda_documentation</li>
    <li>cudaPackages.cuda_gdb</li>
    <li>cudaPackages.cuda_memcheck</li>
    <li>cudaPackages.cuda_nsight</li>
    <li>cudaPackages.cuda_nvcc</li>
    <li>cudaPackages.cuda_nvdisasm</li>
    <li>cudaPackages.cuda_nvml_dev</li>
    <li>cudaPackages.cuda_nvprof</li>
    <li>cudaPackages.cuda_nvprune</li>
    <li>cudaPackages.cuda_nvrtc</li>
    <li>cudaPackages.cuda_nvtx</li>
    <li>cudaPackages.cuda_nvvp</li>
    <li>cudaPackages.cuda_sanitizer_api</li>
    <li>cudatoolkit (cudaPackages.cudatoolkit ,cudatoolkit_11)</li>
    <li>cudaPackages.cudnn (cudaPackages.cudnn_8_4_0)</li>
    <li>cudaPackages.cudnn_8_3_2</li>
    <li>cudaPackages.cutensor</li>
    <li>cudaPackages.fabricmanager</li>
    <li>cudaPackages.libcublas</li>
    <li>cudaPackages.libcufft</li>
    <li>cudaPackages.libcufile</li>
    <li>cudaPackages.libcurand</li>
    <li>cudaPackages.libcusolver</li>
    <li>cudaPackages.libcusparse</li>
    <li>cudaPackages.libnpp</li>
    <li>cudaPackages.libnvidia_nscq</li>
    <li>cudaPackages.libnvjpeg</li>
    <li>cudaPackages.nccl</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cudaPackages.nsight_systems</li>
    <li>cudaPackages.nvidia_fs</li>
    <li>forge</li>
    <li>gpu-burn</li>
    <li>gromacsCudaMpi</li>
    <li>gwe</li>
    <li>katagoWithCuda</li>
    <li>librealsenseWithCuda</li>
    <li>magma</li>
    <li>nvtop</li>
    <li>nvtop-nvidia</li>
    <li>python310Packages.TheanoWithCuda</li>
    <li>python310Packages.pycuda</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python39Packages.TheanoWithCuda</li>
    <li>python39Packages.pycuda</li>
    <li>python39Packages.pynvml</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.torchWithCuda</li>
    <li>xgboostWithCuda</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 185557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>cudaPackages.nvidia_driver</li>
    <li>python310Packages.cupy</li>
    <li>python39Packages.cupy</li>
    <li>truecrack-cuda</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>cudaPackages.tensorrt (cudaPackages.tensorrt_8_4_0)</li>
    <li>ethminer (ethminer-cuda)</li>
    <li>mathematica-cuda</li>
    <li>python310Packages.numbaWithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.tensorrt</li>
    <li>python39Packages.numbaWithCuda</li>
    <li>python39Packages.pyrealsense2WithCuda</li>
    <li>python39Packages.tensorrt</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>colmapWithCuda</li>
    <li>cudaPackages.cuda_cccl</li>
    <li>cudaPackages.cuda_cudart</li>
    <li>cudaPackages.cuda_cuobjdump</li>
    <li>cudaPackages.cuda_cupti</li>
    <li>cudaPackages.cuda_cuxxfilt</li>
    <li>cudaPackages.cuda_demo_suite</li>
    <li>cudaPackages.cuda_documentation</li>
    <li>cudaPackages.cuda_gdb</li>
    <li>cudaPackages.cuda_memcheck</li>
    <li>cudaPackages.cuda_nsight</li>
    <li>cudaPackages.cuda_nvcc</li>
    <li>cudaPackages.cuda_nvdisasm</li>
    <li>cudaPackages.cuda_nvml_dev</li>
    <li>cudaPackages.cuda_nvprof</li>
    <li>cudaPackages.cuda_nvprune</li>
    <li>cudaPackages.cuda_nvrtc</li>
    <li>cudaPackages.cuda_nvtx</li>
    <li>cudaPackages.cuda_nvvp</li>
    <li>cudaPackages.cuda_sanitizer_api</li>
    <li>cudatoolkit (cudaPackages.cudatoolkit ,cudatoolkit_11)</li>
    <li>cudaPackages.cudnn (cudaPackages.cudnn_8_4_0)</li>
    <li>cudaPackages.cudnn_8_3_2</li>
    <li>cudaPackages.cutensor</li>
    <li>cudaPackages.fabricmanager</li>
    <li>cudaPackages.libcublas</li>
    <li>cudaPackages.libcufft</li>
    <li>cudaPackages.libcufile</li>
    <li>cudaPackages.libcurand</li>
    <li>cudaPackages.libcusolver</li>
    <li>cudaPackages.libcusparse</li>
    <li>cudaPackages.libnpp</li>
    <li>cudaPackages.libnvidia_nscq</li>
    <li>cudaPackages.libnvjpeg</li>
    <li>cudaPackages.nccl</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cudaPackages.nsight_systems</li>
    <li>cudaPackages.nvidia_fs</li>
    <li>forge</li>
    <li>gpu-burn</li>
    <li>gromacsCudaMpi</li>
    <li>gwe</li>
    <li>katagoWithCuda</li>
    <li>librealsenseWithCuda</li>
    <li>magma</li>
    <li>nvtop</li>
    <li>nvtop-nvidia</li>
    <li>python310Packages.TheanoWithCuda</li>
    <li>python310Packages.pycuda</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python39Packages.TheanoWithCuda</li>
    <li>python39Packages.pycuda</li>
    <li>python39Packages.pynvml</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.torchWithCuda</li>
    <li>xgboostWithCuda</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>

**Comment by samuela** [comment]
Here are the failure logs:
```
error: builder for '/nix/store/ybsmhdrmjy8xarlvf53kzlj4bbfw7b20-Mathematica_13.1.0_BNDL_LINUX.sh.drv' failed with exit code 1;
       last 8 log lines:
       >
       > ***
       > This nix expression requires that Mathematica_13.1.0_BNDL_LINUX.sh is
       > already part of the store. Find the file on your Mathematica CD
       > and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
       >
       > ***
       >
       For full logs, run 'nix log /nix/store/ybsmhdrmjy8xarlvf53kzlj4bbfw7b20-Mathematica_13.1.0_BNDL_LINUX.sh.drv'.
error: 1 dependencies of derivation '/nix/store/nas8dqxwnya0h3bc5ddnygbzhi1ydjr6-mathematica-cuda-13.1.0.drv' failed to build
error: builder for '/nix/store/yg5akr0m7llg7d4ngjwd9i96jw6sz1qw-TensorRT-8.4.0.6.Linux.x86_64-gnu.cuda-11.6.cudnn8.3.tar.gz.drv' failed with exit code 1;
       last 10 log lines:
       > download the 8.4.0.6 Linux x86_64 TAR package for CUDA 11.7 from
       > https://developer.nvidia.com/tensorrt.
       >
       > Once you have downloaded the file, add it to the store with the following
       > command, and try building this derivation again.
       >
       > $ nix-store --add-fixed sha256 TensorRT-8.4.0.6.Linux.x86_64-gnu.cuda-11.6.cudnn8.3.tar.gz
       >
       > ***
       >
       For full logs, run 'nix log /nix/store/yg5akr0m7llg7d4ngjwd9i96jw6sz1qw-TensorRT-8.4.0.6.Linux.x86_64-gnu.cuda-11.6.cudnn8.3.tar.gz.drv'.
error: 1 dependencies of derivation '/nix/store/9lqc02hrkn62fv9nsd6iw81kdyy55qmj-cudatoolkit-11.7-tensorrt-8.4.0.6.drv' failed to build
error: 2 dependencies of derivation '/nix/store/1qlmd5zrk7l2bm77r0p36qg76mnygcr2-python3.10-tensorrt-8.4.0.6.drv' failed to build
error: 2 dependencies of derivation '/nix/store/7lim89bfbd19k8rbgjcpkkkdh4cysh1r-python3.9-tensorrt-8.4.0.6.drv' failed to build
error: builder for '/nix/store/052hqk4l07mmx9vzyh0zx6ppmgr99iw0-python3.10-numba-0.56.2.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/s69sx2va20j7casl68dsf0b2xjjv0ic2-numba-0.56.2.tar.gz
       > source root is numba-0.56.2
       > setting SOURCE_DATE_EPOCH to timestamp 1662090503 of file numba-0.56.2/numba/_version.py
       > patching sources
       > applying patch /nix/store/hmp6p98pg3gkjcr9nvb5a825dhcch41n-cuda_path.patch
       > patching file numba/cuda/cuda_paths.py
       > Hunk #2 FAILED at 32.
       > Hunk #3 succeeded at 61 with fuzz 2 (offset -10 lines).
       > Hunk #4 succeeded at 77 (offset -10 lines).
       > 1 out of 4 hunks FAILED -- saving rejects to file numba/cuda/cuda_paths.py.rej
       For full logs, run 'nix log /nix/store/052hqk4l07mmx9vzyh0zx6ppmgr99iw0-python3.10-numba-0.56.2.drv'.
error: builder for '/nix/store/h9q30xk65f89k4lnw68gdm3zffkka41g-python3.9-numba-0.56.2.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/s69sx2va20j7casl68dsf0b2xjjv0ic2-numba-0.56.2.tar.gz
       > source root is numba-0.56.2
       > setting SOURCE_DATE_EPOCH to timestamp 1662090503 of file numba-0.56.2/numba/_version.py
       > patching sources
       > applying patch /nix/store/hmp6p98pg3gkjcr9nvb5a825dhcch41n-cuda_path.patch
       > patching file numba/cuda/cuda_paths.py
       > Hunk #2 FAILED at 32.
       > Hunk #3 succeeded at 61 with fuzz 2 (offset -10 lines).
       > Hunk #4 succeeded at 77 (offset -10 lines).
       > 1 out of 4 hunks FAILED -- saving rejects to file numba/cuda/cuda_paths.py.rej
       For full logs, run 'nix log /nix/store/h9q30xk65f89k4lnw68gdm3zffkka41g-python3.9-numba-0.56.2.drv'.
error: builder for '/nix/store/ylafyjflmppbnxq5zjypqhxpdyg39ciz-ethminer-0.19.0.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/5i10bszy2380gn0z1a6d4gf4pyhg51ps-cli11-2.2.0/include/CLI/App.hpp:594:35: note:   no known conversion for argument 2 from 'unsigned int' to 'CLI::callback_t' {aka 'std::function<bool(const std::vector<std::__cxx11::basic_string<char> >&)>'}
       >   594 |                        callback_t option_callback,
       >       |                        ~~~~~~~~~~~^~~~~~~~~~~~~~~
       > /nix/store/5i10bszy2380gn0z1a6d4gf4pyhg51ps-cli11-2.2.0/include/CLI/App.hpp:701:13: note: candidate: 'CLI::Option* CLI::App::add_option(std::string)'
       >   701 |     Option *add_option(std::string option_name) {
       >       |             ^~~~~~~~~~
       > /nix/store/5i10bszy2380gn0z1a6d4gf4pyhg51ps-cli11-2.2.0/include/CLI/App.hpp:701:13: note:   candidate expects 1 argument, 4 provided
       > make[2]: *** [ethminer/CMakeFiles/ethminer.dir/build.make:76: ethminer/CMakeFiles/ethminer.dir/main.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:516: ethminer/CMakeFiles/ethminer.dir/all] Error 2
       > make: *** [Makefile:156: all] Error 2
       For full logs, run 'nix log /nix/store/ylafyjflmppbnxq5zjypqhxpdyg39ciz-ethminer-0.19.0.drv'.
error: builder for '/nix/store/bam31rgddbrr4k4awcv30f2y8yvskiz6-librealsense-2.45.0.drv' failed with exit code 1;
       last 10 log lines:
       >    84 |             self.deleter = [](void* ptr){ delete[] ptr; };
       >       |                                                    ^~~
       > /build/source/wrappers/python/pyrs_internal.cpp: In lambda function:
       > /build/source/wrappers/python/pyrs_internal.cpp:100:52: warning: deleting 'void*' is undefined [-Wdelete-incomplete]
       >   100 |             self.deleter = [](void* ptr){ delete[] ptr; };
       >       |                                                    ^~~
       > [182/219] Building CXX object wrappers/python/CMakeFiles/pyrealsense2.dir/pyrs_sensor.cpp.o
       > [183/219] Building CXX object wrappers/python/CMakeFiles/pyrealsense2.dir/c_files.cpp.o
       > [184/219] Building CXX object wrappers/python/CMakeFiles/pybackend2.dir/pybackend.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/bam31rgddbrr4k4awcv30f2y8yvskiz6-librealsense-2.45.0.drv'.
error: builder for '/nix/store/i5gfqsph1rnff5q674y7wdpf5zc1ql9w-librealsense-2.45.0.drv' failed with exit code 1;
       last 10 log lines:
       > /build/source/wrappers/python/pyrs_internal.cpp:100:52: warning: deleting 'void*' is undefined [-Wdelete-incomplete]
       >   100 |             self.deleter = [](void* ptr){ delete[] ptr; };
       >       |                                                    ^~~
       > [174/219] Building CXX object wrappers/python/CMakeFiles/pyrealsense2.dir/pyrs_frame.cpp.o
       > [175/219] Building CXX object CMakeFiles/realsense2.dir/src/rs.cpp.o
       > [176/219] Building CXX object wrappers/python/CMakeFiles/pyrealsense2.dir/pyrs_sensor.cpp.o
       > [177/219] Building CXX object wrappers/python/CMakeFiles/pyrealsense2.dir/c_files.cpp.o
       > [178/219] Building CXX object wrappers/python/CMakeFiles/pybackend2.dir/pybackend.cpp.o
       > [179/219] Building C object CMakeFiles/realsense2.dir/third-party/sqlite/sqlite3.c.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/i5gfqsph1rnff5q674y7wdpf5zc1ql9w-librealsense-2.45.0.drv'.
error: 9 dependencies of derivation '/nix/store/pa79sf239g5x61lvv6zbv951gyyq8xc0-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/394lrlh4bpkaxnafknlfwadp3qardczh-review-shell.drv' failed to build
```

**Comment by samuela** [comment]
TensorRT and mathematica require special downloads. 

`ethminer`, `numbaWithCuda`, and `pyrealsense2WithCuda ` are all failing on master. We should mark them as broken.

So this change LGTM modulo the formatting changes mentioned above.

**Comment by dguibert** [comment]
I've amending the formatting changes.
Thx for reviewing this.

(it's time to package version 11.8 released few days ago :wink: )

**Review by samuela — APPROVED** [review]

**Comment by samuela** [comment]
Thanks so much for your PR @dguibert, and thank you for your patience getting it merged! Can't believe that 11.8 is already here haha

**Comment by dguibert** [comment]
11.8 update has been proposed in https://github.com/NixOS/nixpkgs/pull/194705

**Pull request merged**